aospy/data_loader.py

```diff
@@ -127,16 +128,54 @@ def _sel_var(ds, var):
     raise LookupError(msg)
 
 
+def _average_time_bounds(ds):
+    """Return the average of each set of time bounds in the Dataset.
+
+    Parameters
+    ----------
+    ds : xarray.Dataset
+        A Dataset containing a time bounds array with name matching
+        internal_names.TIME_BOUNDS_STR.  This time bounds array must have one
+        of two shapes: (1) one dimension, with length 1 longer than the
+        Dataset's time array, or (2) two dimensions, one of which's coordinates
+        is the time array, and the other is length-2.
+
+    Returns
+    -------
+    xarray.DataArray
+        The mean of the start and end times of each timestep in the original
+        Dataset.
+
+    Raises
+    ------
+    NotImplementedError
+        If the time bounds array doesn't match one of the supported shapes
+        specified above.
+
+    """
+    bounds = ds[TIME_BOUNDS_STR]
+    if bounds.ndim == 1:
+        assert len(bounds) == len(ds[TIME_STR]) + 1
+        avg = 0.5*(bounds[:-1] + bounds[1:])
+    elif bounds.ndim == 2:
+        assert set([2, len(ds[TIME_STR])]) == set(bounds.shape)
```

Comment:
Actually this should be:
```python
assert dict(bounds.sizes) == {TIME_STR: ds.sizes[TIME_STR], BOUNDS_STR: 2}
```
because `bounds.sizes` is an `OrderedDict` (so we should cast it as an ordinary `dict` first).

Reply:
@spencerkclark thanks for the review.  All of your suggestions are good; will implement.